File: log-service/services/log_service.go
package services

import (
	"errors"
	"log-service/models"
	"log-service/storage"
)

type LogService struct {
	Storage storage.StorageInterface
}

func NewLogService(storage storage.StorageInterface) LogServiceInterface {
	return &LogService{Storage: storage}
}

func (ls *LogService) GetLogs(ruleID, userID string) ([]models.Log, error) {
	logs, err := ls.Storage.GetLogs(ruleID, userID)
	if err != nil {
		return nil, err
	}
	return logs, nil
}

func (ls *LogService) GetLogByID(id string) (*models.Log, error) {
	log, err := ls.Storage.GetLogByID(id)
	if err != nil {
		return nil, err
	}
	if log == nil {
		return nil, errors.New("log not found")
	}
	return log, nil
}

File: client/src/services/logService.js
import axios from '../api/axios';

const logService = {
  getLogs: async (userId = '', ruleId = '') => {
    const params = new URLSearchParams();
    if (userId) params.append('user_id', userId);
    if (ruleId) params.append('rule_id', ruleId);
    const response = await axios.get(`/logs${params.toString() ? `?${params.toString()}` : ''}`);
    return response.data.data || [];
  },
};

export default logService;
